lib/ansible/modules/cloud/amazon/lambda.py

```diff
@@ -257,10 +256,13 @@ def main():
     required_together = [['s3_key', 's3_bucket'],
                          ['vpc_subnet_ids', 'vpc_security_group_ids']]
 
+    required_if = [['state', 'present', ['runtime', 'handler', 'role']]]
```

Comment:
I'd rather not have another mode... Maybe this is fine since the module is preview. I'm feeling ambivalent. @ryansb @willthames Do you have an opinion? The ec2_asg module is an example of not requiring a full declarative view (launch_config_name only needs to be specified for initial creation and not required for updates), but maybe we should try to avoid emulating that elsewhere?